src/ai/backend/manager/raft/client.py

```diff
@@ -0,0 +1,225 @@
+import asyncio
+from typing import AsyncIterator, Dict, Final, Optional
+
+from attr import dataclass
+from raftify import RaftNode
+
+from ai.backend.manager.raft.state_machine import HashStore, SetCommand
+
+
+@dataclass
+class RaftKVSLockOptions:
+    lock_name: bytes
+    timeout: Optional[float]
+    ttl: Optional[int]
+
+
+class ConnectOptions:
+    def __init__(self) -> None: ...
+    def with_user(self, user: str, password: str) -> "ConnectOptions":
+        return self
+
+    def with_keep_alive(self, interval: float, timeout: float) -> "ConnectOptions":
+        return self
+
+    def with_keep_alive_while_idle(self, enabled: bool) -> "ConnectOptions":
+        return self
+
+    def with_connect_timeout(self, connect_timeout: float) -> "ConnectOptions":
+        return self
+
+    def with_timeout(self, timeout: float) -> "ConnectOptions":
+        return self
+
+    def with_tcp_keepalive(self, tcp_keepalive: float) -> "ConnectOptions":
+        return self
+
+
+class RaftKVSClient:
+    def __init__(
+        self,
+        raft_node: RaftNode,
+        endpoints: list[str],
+        connect_options: Optional["ConnectOptions"] = None,
+    ) -> None:
+        self.raft_node = raft_node
+        self.endpoints = endpoints
+        self.connect_options = connect_options
```

Comment:
I would prefer it if internal fields were not accessed from outside as much as possible, but why are this fields used as public?